src/components/generic/PaginationNavInput.vue

```diff
@@ -0,0 +1,148 @@
+  >
+    <ul
+      class="pl-0 d-flex justify-content-center"
+    >
+      <li
+        :class="{ 'disabled' : prevDisabled}"
+        class="page-item btn-prev pl-0"
+        data-qa="prev button"
+      >
+        <SmartLink
+          :destination="prevUrl"
+          :aria-label="$t('actions.previous')"
+          :disabled="prevDisabled"
+          :aria-hidden="prevDisabled"
+          class="page-link"
+        >
+          <span class="icon-arrow-down" />
+          {{ $t('actions.previous') }}
+        </SmartLink>
+      </li>
+      <li
+        class="page-item page-input"
+      >
+        <b-form-input
+          v-model="page"
+          type="number"
+          :number="true"
+          :aria-label="$t('pageNumber')"
+          :min="1"
+          :max="totalPages"
+          data-qa="pagination input"
+          @change="changePaginationNav"
+        /> {{ $t('of') }} {{ totalPages }}
+      </li>
+      <li
+        :class="{ 'disabled' : nextDisabled}"
+        class="page-item btn-next pr-0"
+        data-qa="next button"
+      >
+        <SmartLink
+          :destination="nextUrl"
+          :aria-label="$t('actions.next')"
+          :disabled="nextDisabled"
+          :aria-hidden="nextDisabled"
+          class="page-link"
+        >
+          {{ $t('actions.next') }}
+          <span class="icon-arrow-down" />
+        </SmartLink>
+      </li>
+    </ul>
+  </nav>
+</template>
+
+<script>
+  import SmartLink from './SmartLink';
+
+  export default {
+    name: 'PaginationNavInput',
+
+    components: {
+      SmartLink
+    },
+
+    props: {
+      perPage: {
+        type: Number,
+        default: 24
+      },
+      totalResults: {
+        type: Number,
+        default: 0
+      },
+      scrollToId: {
+        type: String,
+        default: 'header'
+      },
+      maxResults: {
+        type: Number,
+        default: null
+      }
+    },
+
+    data() {
+      return {
+        page: Number(this.$route?.query?.page) || 1
+      };
+    },
+
+    computed: {
+      totalPages() {
+        const atLeastOne = Math.max(this.totalResults, 1);
+        return Math.ceil(Math.min(atLeastOne, this.maxResults || atLeastOne) / this.perPage);
+      },
+
+      prevDisabled() {
+        return this.page <= 1;
+      },
+
+      prevUrl() {
+        return this.linkGen(this.page - 1);
+      },
+
+      nextUrl() {
+        return this.linkGen(this.page + 1);
+      },
+
+      nextDisabled() {
+        return this.page === this.totalPages;
+      }
+    },
+
+    watch: {
+      '$route.query.page'() {
+        this.page = Number(this.$route?.query?.page) || 1;
+        this.scrollToElement();
+      }
+    },
+
+    methods: {
+      changePaginationNav() {
+        const newRouteQuery = {  ...this.$route.query, page: this.page };
+        const newRoute = { path: this.$route.path, query: newRouteQuery };
+        // TODO: this isn't scrolling to the right place for the search interface.
```

Comment:
The "scrollToId" is "#header" and I think this isn't always working, especially on the search pages, as the header element isn't always visible. Could we replace this by "window.scrollTo(0, 0);" so it always scrolls to the top of the page?